### ISSUE TITLE
Investigate Switch to Dask Expressions

### ISSUE THREAD

**Issue by dougbrn** [lincc-frameworks/tape#382]
Dask Expressions has reached the development stage where dask now has a deprecation warning out for dask dataframes. We should investigate how feasible it is to switch TAPE to use dask expressions.
```
The current Dask DataFrame implementation is deprecated. 
  In a future release, Dask DataFrame will use a new implementation that
  contains several improvements including a logical query planning.
  The user-facing DataFrame API will remain unchanged.
  
  The new implementation is already available and can be enabled by
  installing the dask-expr library:
  
      $ pip install dask-expr
  
  and turning the query planning option on:
  
      >>> import dask
      >>> dask.config.set({'dataframe.query-planning': True})
      >>> import dask.dataframe as dd
  
  API documentation for the new implementation is available at
  https://docs.dask.org/en/stable/dask-expr-api.html
  
  Any feedback can be reported on the Dask issue tracker
  https://github.com/dask/dask/issues 
  
  To disable this warning in the future, set dask config:
  
      # via Python
      >>> dask.config.set({'dataframe.query-planning-warning': False})
  
      # via CLI
      dask config set dataframe.query-planning-warning False
  
  
    import dask.dataframe as dd

```

**Comment by dougbrn** [comment]
I've enabled query planning locally and our unit test suite runs without errors, I still want to verify in more detail how performance is being affected.

**Comment by dougbrn** [comment]
Alright, I've spent a bit more time with this. Unfortunately this is not a "flip a switch and prosper" situation. When originally flipping the config switch, I noticed things were running without issue but I wasn't noticing any difference in behavior/performance. This is because the way we've implemented our EnsembleFrame object is tethered to the original dask dataframe implementation. What this means:
*  We're going to need to re-investigate how we implement the EnsembleFrame based on dask-expr, I initially poked around to see whether it was just a matter of remapping dispatch/backend calls from dataframe to their expr equivalents, but it doesn't appear like there's parity between the APIs on this. dask-expr seems to have refactored things into collections, which is probably our starting point for extending: https://github.com/dask-contrib/dask-expr/blob/8987021e2c7dcf53bcc365e21eaca8b313868978/dask_expr/_collection.py#L3711
* dask-expr is still not the default and the API seems in flux at the moment, it probably makes sense to pin to an older version of dask (2024.1) until dataframe is completely deprecated in favor of dask-expr.

As an aside, with things like this and nested array implementations mounting we're at a crossroads where we could consider leaving TAPE as-is and pursuing a successor package that takes our learnings from this...
